### ISSUE TITLE
Change "say"s to exceptions

### ISSUE THREAD

**Issue by JJ** [noisegul/perl6-p6doc#7]
There are far too many "say"s in the code. They should be changed to exceptions, to be dealt with at an appropriate level. Even if the only thing we are going to be doing is printing them, we'll need to test them and thus we need to use exceptions.

**Comment by JJ** [comment]
I can't open this again, but a failure is not an exception, it's a wrapped exception and the exception will be thrown anyway if it's not properly caught. There are also a few `die`s here and there. 
Perl 6 has got a rich exception set and if there's no proper X:: class you can use X::AdHoc.

**Comment by noisegul** [comment]
Thanks for the clarification, I will open another issue for this once the branch merge is done to outline this in more detail.
